internal/primitives/state-machine/backend/in_memory_backend.go

```diff
@@ -24,9 +23,34 @@ func NewMemoryDBTrieBackend[H runtime.Hash, Hasher runtime.Hasher[H]]() MemoryDB
 	}
 }
 
+func NewMemoryDBTrieBackendFromMap[H runtime.Hash, Hasher runtime.Hasher[H]](
```

Comment:
given this is only used in tests, should we not just put this in `overlayed_changes_test.go` to reduce the number of diffs?

Reply:
Oh I see.